### ISSUE TITLE
Images backed up from iOS don't keep correct rotation

### ISSUE THREAD

**Issue by j-peeters** [immich-app/immich#8355]
### The bug

Backup photos to my private immich server (which is running on Unraid in the community Immich docker) don't keep their orientations compared to how they are displayed in the Immich app. 
In the iOS Immich app, all pictures show in the right orientation. 
![IMG_3929](https://github.com/immich-app/immich/assets/4502360/d79d89ce-2e01-416c-9e4b-a6538e2b6e3f)

On the web UI though, most photos are not shown in the same way:
![20240329 Photos - Immich](https://github.com/immich-app/immich/assets/4502360/16a72559-5ff9-457b-a765-5949e2ee0423)

I've uploaded the photos through a Cloudflared tunnel from my iPhone to my server. As far as I know this is an encrypted tunnel and no metadata is stripped.

I have run the extract metadata post upload but that didn't help.
Thanks for any ideas. I'd love to help troubleshoot this but some info below is lacking because of limited knowledge of docker. (the reason I use Unraid ;-))

### The OS that Immich Server is running on

Unraid 6.12.9

### Version of Immich Server

1.100.0

### Version of Immich Mobile App

1.100.0 build 146

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
I have no idea where to find this info in Unraid. If someone can help me locate it I'll post it. Thanks.
```

### Your .env content

```Shell
I have no idea where to find this info in Unraid. If someone can help me locate it I'll post it. Thanks.
```

### Reproduction steps

```bash
1.start a backup from iOS app
2.view images in Immich server
```

**Comment by alextran1502** [comment]
Can you share this picture for troubleshooting? 
![image](https://github.com/immich-app/immich/assets/27055614/37bd81b9-83e8-4a09-b360-e234409151e6)

**Comment by j-peeters** [comment]
Of course, here is the file. It is the original, a HEIC file (the now default file format for photos in iOS). 
If I export it from Apple Photos as a JPEG it exports the image with the correct orientation. So the embedded EXIF data get normalized. I hope this file is workable. Thanks for looking into this!

Orientation data in HEIC:
`Orientation: Rotate 180`

Orientation data in JPEG:
`Orientation: Horizontal (normal)`

[IMG_3925.HEIC.zip](https://github.com/immich-app/immich/files/14805766/IMG_3925.HEIC.zip)

**Comment by azhiltsov** [comment]
I Have just installed the Immich on Unraid and observing the same issue with some of my photos (mostly selfies).
Please let me know if I can help with troubleshooting.

**Comment by scrampker** [comment]
A simple rotate/crop tool would really help with similar issues.

**Comment by j-peeters** [comment]
Hi @alextran1502 did the images I shared give an indication as to why Immich is displaying them with the wrong rotation? Thanks.

**Comment by alextran1502** [comment]
@j-peeters Hello; I did look at the orientation info extracted from the photo and I don't see anything abnormal to other images. At the moment, I haven't figured out the cause yet. We might need to look into updating the dependency such as Sharp to see if it help generating the correct orientation

**Comment by j-peeters** [comment]
Thanks for the update @alextran1502 hopefully an updated Sharp will fix it. Would love to use this fabulous service. Thanks for your time and effort!

**Comment by j-peeters** [comment]
Hi @alextran1502, I'm eager to start using Immich, so I wonder if you have had time to see if an updated dependency solves the orientation issue? Thanks.

**Comment by jrasm91** [comment]
Turns out HEIF exif is a bit weird and rotation is intentionally ignored by vips. I think it isn't actually in the spec for HEIF.

**Comment by j-peeters** [comment]
> Turns out HEIF exif is a bit weird and rotation is intentionally ignored by vips. I think it isn't actually in the spec for HEIF. 

Thanks for the comment @jrasm91. Do you think that you could implement something in Immich that helps show images (and add I understand videos) in the orientation in which they were taken? Because otherwise it's not a great way to view your material if it's shown upside down or sideways. Curious to read what you think.

**Comment by jrasm91** [comment]
Yeah I was considering adding something in immich to basically always use the rotation from exif if available.

**Comment by j-peeters** [comment]
> Yeah I was considering adding something in immich to basically always use the rotation from exif if available.

That's really great. Looking forward to it! Thanks in advance.

**Comment by mertalev** [comment]
From what I understand, rotation is handled as a HEIF transformation, so the EXIF orientation is supposed to just be informational. If you rotate based on the tag, you'll have double rotated the image. I think the issue in this case is that the image has an EXIF tag, but (incorrectly) doesn't have a transformation to rotate it.

**Comment by TheRedCyclops** [comment]
I'm having a very similar issue, in the app it seems to recognize the EXIF orientation and it correctly orients the picture. But on the web interface it does not, even when rotated in the app

**Comment by dionorgua** [comment]
KDE Gwenview, GIMP displays same picture incorrectly.
Digikam is OK

**Comment by kalix123** [comment]
Any updates on this? I just discovered immich and would like to keep using it. But this features makes a lot of my photos rotated the wrong way.

**Comment by SoulGit** [comment]
Same experience here.  Issue seems to occur only with images taken with a 3rd party camera app (never experienced this problem previously until I started using a new camera app last week.) Additionally, the problem extends beyond the web view: Images taken on the iPhone display correctly within Immich on the iPhone, same images are rotated left in the web UI and also when viewed on an iPad via Immich app.  

Some additional info: In the instances where this occurs, the jpg has the proper orientation, the HEIC does not.  It looks like the Orientation tag is present, but the Rotation tag is not, so perhaps the Immich renderer gets confused and the HEIC image doesn't get rotated properly?

**Comment by j-peeters** [comment]
Yesterday I reinstalled Immich to see if this issue was resolved. Unfortunately it still existed. I understand you all are busy but hopefully someone can have a look at it again and maybe find a way to get these images displayed correctly. My neck would love it 🙃

**Comment by wisekiller** [comment]
Same here. HEIC photos from iCloud are still showing up sideways or upside down on the latest version. Looking forward to either a permanent fix for the metadata processing or the upcoming manual rotation feature!

**Comment by YtachY** [comment]
# Not only in iOS, it's HEIC/HEIF format.

I am writing my case here clearly and seeking for any helps: 

### Summary:
I got a bunch of HEIC files in the phone (which saved for years before I have my own immich server deployed). If I want it to visualize photos correctly, there are some ways:

1. Immich development team update (sth) and the photos appear correctly.
2. Or I convert all .HEIC files to JPG files and then backup to immich server.
3. Or I backup all .HEIC files then rotate it manually.
4. From now on, I don't use any .HEIC file until the 1st have completed.

### Anyone can help with 2,3,4 please let me know.

# Demo (video)

### Env & behaviour

- Immich Android v2.7.4. 
- Immich server: v2.5.6.
- After backed up the HIEC photo into server, in any other client see it already rotated 90 degree (Web, iOS...)
- Even I move the file to other client, the "auto rotate" behaviour still exist.

### Video
https://drive.google.com/file/d/1MXOZRl6dBofmfdA39rXGvCkrwpDPhwj_/view?usp=drive_link

### Image

<img width="555" height="1080" alt="Image" src="https://github.com/user-attachments/assets/e671d02b-35f9-40f8-ab4c-29c568689d63" />


<img width="1871" height="957" alt="Image" src="https://github.com/user-attachments/assets/7aaf45a3-62e2-4fbf-a23b-140c3684de6e" />


# If you need more materials/debug, I'd love to provide it.

**Comment by YtachY** [comment]
### Another findings

I have just found out that the "auto rotate" of .HEIC only affects the potrait photos, which has height larger than width.

For the landscape's photos it remains correct rotation without any "auto".

**Comment by jrasm91** [comment]
Just to provide an update on this - the HEIF/HEIC format implements rotation differently than all other photos. It is stored as a filter/transform. In the specification for the image format it actually explicitly says the exif.orientation should _not_ be used. Unfortunately the industry has used this field for rotation/orientation for decades and most editors know about it and use whatever value it has. Now with a new image format that says "don't use that field for rotation" we have this issue where some viewers know to ignore it for HEIF/HEIC while others don't. To add to the confusion, I have heard of iPhone's themselves taking pictures and double rotating them.

So I guess the TLDR is that HEIF/HEIC made a whole mess of this by changing the usage of exif.orientation and now our choices are either:
1. Follow the spec and don't double rotate (some set of iPhone pictures continue to show up wrong)
2. Ignore the spec and  double rotate (another set of iPhone pictures continue to show up wrong)

I think the only sensible way forward is to continue to implement the spec and wait for other tools to use it correctly.

It's worth noting that the editor feature has been released on the web. There is a button in the asset viewer to go to the editor, which has rotation buttons. I believe you will be able to fix rotation issues with this feature.

<img width="523" height="175" alt="Image" src="https://github.com/user-attachments/assets/8170bbb5-25da-4ac2-a9a2-ba71b6d66254" />

<img width="424" height="231" alt="Image" src="https://github.com/user-attachments/assets/fc5e65f5-205f-4afe-a4ea-6c9e35fc3372" />

**Comment by frejaya** [comment]
For what it's worth, as a workaround on iPhone, I switched to using the ProShot app (and I don't use the HEIC format) to take photos and I just never use the Apple camera app anymore. Turns out I actually like ProShot a lot better so this has worked for me. But I have other family on my Immich who just use the Apple camera app and their landscape photos still look wrong in Immich.

**Comment by j-peeters** [comment]
@frejaya What file format does ProShot use? If it's JPEG instead of HEIC that would explain that your problem isn't there for you.

**Comment by frejaya** [comment]
> [@frejaya](https://github.com/frejaya) What file format does ProShot use? If it's JPEG instead of HEIC that would explain that your problem isn't there for you.

It's a setting the app offers, which is why I chose it. I set it to use JPEG, but it allows you to choose from JPEG, RAW, HEIC or a combination.
